pySDC/implementations/convergence_controller_classes/adaptivity.py

```diff
@@ -212,6 +212,55 @@ def get_local_error_estimate(self, controller, S, **kwargs):
         return S.levels[0].status.error_embedded_estimate
 
 
+class AdaptivityRK(Adaptivity):
+    '''
+    Adaptivity for Runge-Kutta methods. Basically, we need to change the order in the step size update
+    '''
+    def check_parameters(self, controller, params, description):
+        '''
+        Check whether parameters are compatible with whatever assumptions went into the step size functions etc.
+        For adaptivity, we need to know the order of the scheme.
+
+        Args:
+            controller (pySDC.Controller): The controller
+            params (dict): The params passed for this specific convergence controller
+            description (dict): The description object used to instantiate the controller
+
+        Returns:
+            bool: Whether the parameters are compatible
+            str: The error message
+        '''
+        if 'update_order' not in params.keys():
+            return False, 'Adaptivity needs an order for the update rule! Please set some up in \
+description[\'convergence_control_params\'][\'update_order\']!'
+
+        return super(AdaptivityRK, self).check_parameters(controller, params, description)
+
+    def get_new_step_size(self, controller, S):
+        '''
+        Determine a step size for the next step from an embedded estimate of the local error of the current step.
+
+        Args:
+            controller (pySDC.Controller): The controller
+            S (pySDC.Step): The current step
+
+        Returns:
+            None
+        '''
+        # check if we performed the desired amount of sweeps
+        if S.status.iter == S.params.maxiter:
+            L = S.levels[0]
+
+            # compute next step size
+            order = self.params.update_order
+            e_est = self.get_local_error_estimate(controller, S)
+            L.status.dt_new = self.compute_optimatal_step_size(self.params.beta, L.params.dt, self.params.e_tol, e_est,
```

Comment:
`optimatal`...?

Reply:
Good catch! It's quite ironic that I made a typo in the commit message where I meant to say "fixed typo" :D